### ISSUE TITLE
Currency rate download times out

### ISSUE THREAD

**Issue by camillehuot** [moneymanagerex/android-money-manager-ex#1753]
Hi , I am a new user of mmex android, I like it very much.

But when I try to update the currency rates, I got a message "Error: timeout".

I only have Euro (main) and Indian rupees.

All rates stay at 1 (default) and are not updated.

Can someone confirm there is an issue?
Or I do something wrong.

Thanks
Camille

**Device Information:**
 - Device: OnePlus 10 pro
 - OS: android 14
 - Storage: sync Google drive
 - Synchronization: start+10 min
 - Encrypted: no
 - App Version: 2024.08.01

**Additional context**

**Comment by guanlisheng** [comment]
related to #1666

**Comment by camillehuot** [comment]
Thanks for the reply

After browsing the code, my understanding from this file: https://github.com/moneymanagerex/android-money-manager-ex/blob/master/app/src/main/java/com/money/manager/ex/currency/ExchangeRateUpdaterFactory.java#L35

Is that the current service is **Fixer**. I guess it does not work because Fixer requires a key since 2018?

May I suggest that users can get a Fixer key (it is free) and set it in the Currency settings?

Or eventually the features associated to rate synchronisation/upload can be commented/removed until another API is found?

**Comment by guanlisheng** [comment]
have no time to dig into it yet while your suggestion sounds reasonable. still, some development work is required.

as a workaround,  users can sync the currency rate via MMEX Destkop as the database will be syned between MMEX Android and MMEX Desktop.

**Comment by guanlisheng** [comment]
hi @camillehuot 
can you help to test it out?
not been released yet but is available from 
 https://drive.google.com/file/d/1V7OYT2_wf0qbII1q81Noq6XZnPCbu4hq/view?usp=sharing

**Comment by camillehuot** [comment]
Hi @guanlisheng 
I tested your APK and the currency update works for me! Thank you very much

Also, while I have no clue about Android dev, I had installed the Android Studio and were able to build the app with your last commit. But I could not deploy the APK as the installer thinks it is a different app with same name and refuses to update. I assume this is because I don't have the proper signing key, or is there something else to configure?
